docs/es/week12/12-2.md

```diff
@@ -0,0 +1,323 @@
+
+Resumen: Continúe la búsqueda por haces hasta que todas las $k$ hipótesis produzcan un token final o hasta que se alcance el límite máximo de decodificación T.
+
+
+### Muestreo
+
+Puede que no queramos la secuencia más probable. En su lugar, podemos tomar muestras de la distribución del modelo.
+
+Sin embargo, el muestreo de la distribución del modelo plantea su propio problema. Una vez que se muestrea una "mala" elección, el modelo se encuentra en un estado que nunca enfrentó durante el entrenamiento, lo que aumenta la probabilidad de una "mala" evaluación continua. Por lo tanto, el algoritmo puede atascarse en ciclos de retroalimentación horribles.
+
+
+### Muestreo de Top-K
+
+Una técnica de muestreo pura en la que se trunca la distribución al $ k $ best y luego se vuelve a normalizar y se muestra la distribución.
+
+<centro>
+<img src = "{{site.baseurl}}/images/week12/12-2/Top_K_Sampling.png" width = "60%" /> <br>
+<b> Fig. 2 </b>: muestreo superior K
+</center>
+
+
+#### Pregunta: ¿Por qué el muestreo Top-K funciona tan bien?
+
+Esta técnica funciona bien porque esencialmente intenta evitar caerse de la variedad del buen lenguaje cuando probamos algo malo usando solo la cabeza de la distribución y cortando la cola.
+
+
+## Evaluación de la generación de texto
+
+La evaluación del modelo de lenguaje requiere simplemente el logaritmo de la verosimilitud de los datos retenidos. Sin embargo, es difícil evaluar texto. Normalmente se utilizan métricas de superposición de palabras con una referencia (BLEU, ROUGE, etc.), pero tienen sus propios problemas.
+
+
+## Modelos de secuencia a secuencia
+
+
+### Modelos de lenguaje condicional
+
+Los modelos de lenguaje condicional no son útiles para generar muestras aleatorias de inglés, pero son útiles para generar un texto dado una entrada.
+
+Ejemplos:
+
+- Dada una oración en francés, generar la traducción al inglés
+- Dado un documento, generar un resumen
+- Dado un diálogo, generar la siguiente respuesta
+- Dada una pregunta, genera la respuesta
+
+
+### Modelos de secuencia a secuencia
+
+Generalmente, el texto de entrada está codificado. Esta incrustación resultante se conoce como un "vector de pensamiento", que luego se pasa al decodificador para generar tokens palabra por palabra.
+
+<centro>
+<img src = "{{site.baseurl}}/images/week12/12-2/s2s_Models.png" width = "60%" /> <br>
+<b> Fig. 3 </b>: Vector de pensamiento
+</center>
+
+
+### Transformador de secuencia a secuencia
+
+La variación de secuencia a secuencia de transformadores tiene 2 pilas:
+
+1. Pila de codificadores: la atención personal no está enmascarada, por lo que cada token en la entrada puede mirar todos los demás token en la entrada
```

Comment:
I wonder if we should use "símbolo" instead of "token" here and elsewhere in the Spanish translation.

Reply:
Will leave as is until it's resolved in Slack. 

Mrore:
IMHO "Simbolo" is intuitively associated with individual characters and using it means "bothering" the readers with an abstraction... should we keep on looking for a better replacement? My guess is it depends on the global philosphy prevails ("LATAM-style" vs "Spain-Style", the former being "simbolo" and the latter "token"). Thanks for your time!